test/crltest.c

```diff
@@ -239,7 +240,11 @@ static int Verify(X509 *leaf, STACK_OF(X509) *roots, STACK_OF(X509_CRL) *crls,
     X509_STORE_CTX_set0_param(ctx, param);
 
     ERR_clear_error();
-    if (X509_verify_cert(ctx) != 1) {
+    retcode = X509_verify_cert(ctx);
+    /* Unset the CA and CRL stacks so they don't get free'd by us. */
+    X509_STORE_CTX_trusted_stack(ctx, NULL);
+    X509_STORE_CTX_set0_crls(ctx, NULL);
```

Comment:
I'm not seeing where either of these fields are freed by `X509_STORE_CTX_free`, though perhaps I'm just not seeing it.

Anyway, this doesn't make sense. One would expect that either both `X509_STORE_CTX_set0_crls` and `X509_STORE_CTX_free` free the existing value or neither to. 

That is, if `ctx` owns `crls` as a result of line 235, then `ctx` is responsible for freeing it in both `set0_crls` and `free`. If `ctx` doesn't own `crls`, then `ctx` should not free it in either function. (At least from inspection, it seems to be the latter.)